packages/components/src/drop-zone/types.ts

```diff
@@ -26,4 +26,9 @@ export type DropZoneProps = {
 	 * It receives the HTML being dropped as an argument.
 	 */
 	onHTMLDrop?: ( html: string ) => void;
+	/**
+	 * The function is called when dropping media into the `DropZone`.
+	 * It receives the media being dropped as an argument.
+	 */
+	onMediaDrop?: ( media: string ) => void;
```

Comment:
I'm not sure about the idea of this new `onMediaDrop` event handler in the components package. It feels like something that's handling a very specific block editor use case. It still receives an HTML string, which might be confusing. Some consumers of the component might be confused about why it doesn't work when they're dragging media items, and it'd take some digging to find they have to set the `dataTransfer` type to `media`.

Looking at precedence, this component doesn't have an `onBlocksDrop`, that's handled via `onDrop` (in `useBlockDropZone`, and this custom dataTransfer type could work similarly?

Reply:
Hmm do you mean leveraging onDrop for the media type instead of adding a custom handler? I can give it a go!

Reply:
I'm not really sure what the right value is for the dataTransfer type but using `onDrop` in this way looks better, thanks for changing it!

Reply:
Yeah `media` is nice because it's explicit, but here we just need something to distinguish between media and non-media; whatever value we give it, DropZone will interpret it as `default` anyway.